apps/rand.c

```diff
@@ -95,8 +97,85 @@ int rand_main(int argc, char **argv)
     argc = opt_num_rest();
     argv = opt_rest();
     if (argc == 1) {
-        if (!opt_long(argv[0], &num) || num <= 0)
+        int factoridx = 0;
+        int shift = 0;
+
+        /*
+         * special case for requesting the max allowed
+         * number of random bytes to be generated
+         */
+        if (!strcmp(argv[0], "max")) {
+            /*
+             * 2^61 bytes is the limit of random output
+             * per drbg instantiation
+             */
+            scaled_num = SIZE_MAX >> 3;
```

Comment:
This depends on the size of size_t.  For a 32 bit size_t, the maximum will be roughly 536 million which isn't all that large.

What about `sizeof(size_t) == 4 ? SIZE_MAX : ((size_t)1 << 61 - 1)`?
Or some preprocessor conditioning?  I'm not sure what it would condition on `THIRTY_TWO_BIT` perhaps.

Anyway, just a thought for consideration.

Reply:
Why should we use size_t for scaled_num?? Let's use uint64_t for it instead.

Reply:
Yeah, uint64_t avoids the problem entirely, lets go with that.